services/provider/server/server.go

```diff
@@ -6,65 +6,214 @@ import (
 	"net"
 	"os"
 
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
 	"github.com/red-hat-storage/ocs-operator/services/provider/common"
 	pb "github.com/red-hat-storage/ocs-operator/services/provider/pb"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/status"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	TicketAnnotation = "ocs.openshift.io/provider-onboarding-ticket"
 )
 
 type ocsProviderServer struct {
 	pb.UnimplementedOCSProviderServer
+	client    client.Client
+	consumer  *ocsConsumer
+	namespace string
+}
+
+func NewOCSProviderServer(ctx context.Context, namespace string) (*ocsProviderServer, error) {
+	client, err := newClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new client. %v", err)
+	}
+
+	consumer, err := NewOCSConsumer(ctx, client)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new OCSConumer instance. %v", err)
+	}
+
+	return &ocsProviderServer{
+		client:    client,
+		consumer:  consumer,
+		namespace: namespace,
+	}, nil
 }
 
 // OnboardConsumer RPC call to onboard a new OCS consumer cluster.
-func (c *ocsProviderServer) OnboardConsumer(ctx context.Context, req *pb.OnboardConsumerRequest) (*pb.OnboardConsumerResponse, error) {
+func (s *ocsProviderServer) OnboardConsumer(ctx context.Context, req *pb.OnboardConsumerRequest) (*pb.OnboardConsumerResponse, error) {
 	mock := os.Getenv(common.MockProviderAPI)
 	if mock != "" {
 		return mockOnboardConsumer(common.MockError(mock))
 	}
-	return &pb.OnboardConsumerResponse{}, nil
+
+	// Validate capacity
+	capacity, err := resource.ParseQuantity(req.Capacity)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "%q is not a valid storageConsumer capacity: %v", req.Capacity, err)
+	}
+
+	// Validate onboardingTicket
+	isTicketUsed := s.consumer.IsTicketUsed(req.OnboardingTicket)
+	if isTicketUsed {
+		return nil, status.Errorf(codes.InvalidArgument, "onboarding ticket is already used by an existing storageConsumer resource")
+	}
```

Comment:
There should be a couple of steps here:
1. Split the ticket into the encrypted and non-encrypted section
1. Validate the ticket using the provided key (and the two sections) 
2. Unmarsh the non-encrypted ticket into a struct (to be provided by @jarrpa)
3. Check the expiry field to make sure the ticket is not expired 
4. Validate that the ticket is not in use - would suggest saving only the encrypted side on the annotation or the sig field

You can encapsulate the logic in a function, something like `validateTicket`